### DISCUSSION TITLE
Ability to group related settings

### DISCUSSION THREAD

**Discussion by jdbit** [fabiocaccamo/django-extra-settings#29] [Ideas]
Hi,
Thank you for the nice app! It would be great to have a possibility to group related settings somehow, for example place them to different tabs or at least place some dividers between a group of fields/settings.

**Comment by fabiocaccamo** [comment]
Hi @jdbit, 
nice suggestion!

I'm having the same need in a project with many settings, I prefixed all settings with their own context group to be able to search them quickly.

A simple and effective solution that comes to my mind (that does not impact the manual management of settings nor the database), is to add a list filter that group settings by their name prefix. The groups would be auto-detected by the application splitting the settings names on the first `_` and considering as groups only prefixes that match multiple settings.

For example, having these settings: 
- `HEADER_TITLE`
- `HEADER_SUBTITLE`
- `HEADER_IMAGE`
- `BODY_TEXT`
- `FOOTER_EMAIL`
- `FOOTER_PHONE`

The list filter would show two options:
- `HEADER`
- `FOOTER`

I like this solution, do you?

**Comment by jdbit** [comment]
Great, thank you!

**Comment by fabiocaccamo** [comment]
Hi @jdbit, 
this feature is available by upgrading the library to `0.5.0` version and configuring the setting `EXTRA_SETTINGS_SHOW_NAME_PREFIX_LIST_FILTER = True`.

**Reply by jdbit** [reply]
Thank you so much for the update!

**Reply by fabiocaccamo** [reply]
Let me know how does it work for your use case.

**Reply by jdbit** [reply]
I've tested it a bit and it works just fine, thank you! I noticed that there is a description field in the admin list view, but I can't change it when create or update the option. It's a bug or feature?:)

**Reply by fabiocaccamo** [reply]
@jdbit you should be able to edit setting description from its detail view, just click on the setting name in the changelist.

**Reply by jdbit** [reply]
you are right, I thought there is no such field, not sure how I missed it:) Thanks!
